### ISSUE TITLE
Remember disabled plugins

### ISSUE THREAD

**Issue by Dragon8oy** [r-clancy/PlugMan#78]
When a plugin is disabled, I think it should be disabled and unloaded every time the server reloads and restarts, and restored to normal behaviour then it is enabled again.
Either this, or a blacklist of plugins could be added to the config, which are automatically unloaded on server start.

This would save time for server admins and stop them having to remember to disable a specific plugin every time.

**Comment by kennyrkun** [comment]
If possible, stop the plugin from loading entirely rather than waiting for it to load and then disable it again.

**Comment by r-clancy** [comment]
> Either this, or a blacklist of plugins could be added to the config, which are automatically unloaded on server start.
> This would save time for server admins and stop them having to remember to disable a specific plugin every time.

PlugMan's disable/unload command is meant to be temporary - if the server admin wishes to have the plugin actually removed, they can simply remove the JAR file from the `plugins` folder. I don't think PlugMan should make permanent changes to the plugins, this should be reserved for those with access to the files.

**Comment by Dragon8oy** [comment]
Well would a blacklist be fine? You'd have to have access to the config to blacklist them, so worrying about access to files shouldn't be an issue with a blacklist.

**Comment by r-clancy** [comment]
> Well would a blacklist be fine? You'd have to have access to the config to blacklist them, so worrying about access to files shouldn't be an issue with a blacklist.

The blacklist idea doesn't make sense to me - if you have to edit the config file to automatically disable the plugins, why not just move the plugin file? Removing the file so it doesn't load at all, rather than disabling as soon as the server starts, is much cleaner. The unload command in particular is a hack as Bukkit doesn't directly expose a way to unload plugins.

**Comment by Dragon8oy** [comment]
Fair enough, if you have to edit a config you may as well move the file instead if t's cleaner
